### PR TITLE
PR: `tests/unit/test_source`: cleanup, parametrize, and increase test coverage

### PR DESCRIPTION
**Pull request by bigtedde**
### Description
In this PR, I cleaned up the `test_source` module by splitting the `paste_text()` testing into its own separate test. I then parametrized `test_paste_text()` to cover more use/edge cases. Finally, I added in tests for removing a source from the sources table, as well as testing the source update button.

### Related Issue
#1754 

### Motivation and Context
One of my Google Summer of Code projects involves cleaning up the Vorta tests, reducing duplicated code, and increasing code coverage.

### How Has This Been Tested?
Tests work locally by running `pytest`, and they pass the GitHub CI checks.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] I have read the [CONTRIBUTING](https://vorta.borgbase.com/contributing/) guide.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

*I provide my contribution under the terms of the [license](./../../LICENSE.txt) of this repository and I affirm the [Developer Certificate of Origin][dco].*

[dco]: https://developercertificate.org/

### CODE REVIEW

File: tests/unit/test_source.py [hunk 1]
```diff
@@ -1,22 +1,94 @@
...
 
-    # Wait for directory sizing to finish
-    qtbot.waitUntil(lambda: len(qapp.main_window.sourceTab.updateThreads) == 0, **pytest._wait_defaults)
+    if valid:
+        # valid paths will be added as a source
+        assert not hasattr(tab, '_msg')
+        qtbot.waitUntil(lambda: tab.sourceFilesWidget.rowCount() == 2, **pytest._wait_defaults)
+        assert tab.sourceFilesWidget.rowCount() == 2
+        # Wait for directory sizing to finish
+        qtbot.waitUntil(lambda: len(qapp.main_window.sourceTab.updateThreads) == 0, **pytest._wait_defaults)
```
Comment (bigtedde) [inline tests/unit/test_source.py:70]: This line is causing problems with the linux tests *most* of the time, but not always. Never seems to have trouble on darwin. Might need to do more digging here. @m3nu @real-yfprojects have either of y'all encountered errors caused by `updateThreads` before?

Specifically, `updateThreads` is not not getting set back to 0 and the wait times out.

File: tests/unit/test_source.py [hunk 2]
```diff
@@ -1,22 +1,92 @@
...
+        main.tabWidget.setCurrentIndex(1)
+        tab = main.sourceTab
+        qtbot.waitUntil(lambda: tab.sourceFilesWidget.rowCount() == 1, timeout=2000)
+        return main, tab
 
+    main, tab = setup()
+    yield main, tab
+
+    # Wait for directory sizing to finish
+    qtbot.waitUntil(lambda: len(qapp.main_window.sourceTab.updateThreads) == 0, timeout=2000)
```
Comment (bigtedde) [inline tests/unit/test_source.py:21]: After some digging, I discovered this wait needs to be done after each source test that involves `updateThreads`. I added this to the `source_env` fixture to happen at teardown.

File: tests/unit/test_source.py (hunk 1 above)
Comment (bigtedde) [reply tests/unit/test_source.py]: Resolved - this needs to be called after each test. Moved to fixture teardown.

**Review by m3nu — DISMISSED** [review]
